pkg/proxy/winkernel/proxier_test.go

```diff
@@ -860,92 +996,2492 @@ func TestEndpointSlice(t *testing.T) {
+
+	svcPortName := proxy.ServicePortName{
+		NamespacedName: makeNSN("ns1", "svc1"),
+		Port:           "p80",
+	}
+
+	// Act
+	makeServiceMap(proxier,
+		makeTestService(svcPortName.Namespace, svcPortName.Name, func(svc *v1.Service) {
+			svc.Spec.ClusterIP = svcIP
+			svc.Spec.Ports = []v1.ServicePort{{
+				Name:     svcPortName.Port,
+				Port:     int32(svcPort),
+				Protocol: v1.ProtocolTCP,
+			}}
+		}),
+	)
+
+	proxier.setInitialized(true)
+	proxier.syncProxyRules()
+
+	// Assert
+	svc := proxier.serviceMap[svcPortName]
+	_, ok := svc.(*serviceInfo)
+	//The ClusterIp should be rejected so the casting to serviceInfo should not be possible
+	if ok {
+		t.Errorf("Service ns1/svc1:p80 shoud be REJECTED. Unexpected behaviour!")
+	}
+
+	//there should not be created any hcn instances because the service was rejected
+	LoadBalancers, err := testHNS.hcninstance.ListLoadBalancers()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if len(LoadBalancers) != 0 {
+		t.Errorf("Service ns1/svc1:p80 shoud be REJECTED. Unexpected behaviour!")
+	}
+}
+
+// TestLoadBalancerReject creates a non-valid LoadBalancer service and verifies if it is
+// successfully rejected
+func TestLoadBalancerReject(t *testing.T) {
+	// Arrange
+	testHNS := NewHCNUtils(&hcntesting.FakeHCN{})
+	proxier := NewFakeProxier(testHNS, NETWORK_TYPE_OVERLAY)
+
+	svcPortName := proxy.ServicePortName{
+		NamespacedName: makeNSN("ns1", "svc1"),
+		Port:           "p80",
+		Protocol:       v1.ProtocolTCP,
+	}
+	svcSessionAffinityTimeout := int32(10800)
+
+	// Act
+	makeServiceMap(proxier,
+		makeTestService(svcPortName.Namespace, svcPortName.Name, func(svc *v1.Service) {
+			svc.Spec.Type = "LoadBalancer"
+			svc.Spec.ClusterIP = svcIP
+			svc.Spec.Ports = []v1.ServicePort{{
+				Name:     svcPortName.Port,
+				Port:     int32(svcPort),
+				Protocol: v1.ProtocolTCP,
+				NodePort: int32(svcNodePort),
+			}}
+			svc.Spec.HealthCheckNodePort = int32(svcHealthCheckNodePort)
+			svc.Status.LoadBalancer.Ingress = []v1.LoadBalancerIngress{{
+				IP: svcLBIP,
+			}}
+			svc.Spec.ExternalTrafficPolicy = v1.ServiceExternalTrafficPolicyTypeLocal
+			svc.Spec.SessionAffinity = v1.ServiceAffinityClientIP
+			svc.Spec.SessionAffinityConfig = &v1.SessionAffinityConfig{
+				ClientIP: &v1.ClientIPConfig{TimeoutSeconds: &svcSessionAffinityTimeout},
+			}
+		}),
+	)
+
+	proxier.setInitialized(true)
+	proxier.syncProxyRules()
+
+	// Assert
+	svc := proxier.serviceMap[svcPortName]
+	svcInfo, ok := svc.(*serviceInfo)
+	if !ok {
+		t.Errorf("Failed to cast serviceInfo %q", svcPortName.String())
+
+	} else { //service has no endpoints so the lb policy should not have been applied
+		if svcInfo.policyApplied {
+			t.Errorf("Service ns1/svc1:p80 has no endpoint information available, but policies are applied. Unexpected behaviour!")
+		}
+	}
+
+	//no LoadBalancers should have been created so this should fail
+	_, err := testHNS.hcninstance.GetLoadBalancerByID(guid)
+
+	if err == nil {
+		t.Error(err)
```

Comment:
This doesn't really make sense. If it's ``nil``, we can't really log it / use it as a test failure. In this case, the test failure should be that we were not expecting the load balancer to exist.